pkg/machine/convert.go

```diff
@@ -278,3 +284,17 @@ func GetAPIV2NodeCloudSpec(machineSpec clusterv1alpha1.MachineSpec) (*apiv1.Node
 
 	return cloudSpec, nil
 }
+
+func extractSpotInstanceConfigs(config *aws.RawConfig) (*string, *string, *bool) {
+	if config.IsSpotInstance != nil && *config.IsSpotInstance {
+		if config.SpotInstanceConfig != nil {
+			return &config.SpotInstanceConfig.MaxPrice.Value,
+				&config.SpotInstanceConfig.InterruptionBehavior.Value,
+				&config.SpotInstanceConfig.PersistentRequest.Value
```

Comment:
Are we sure that this is not going to throw NPEs if some property is `nil`? For example, if `MaxPrice` is `nil`, accessing `MaxPrice.Value` will throw the NPE.

Reply:
That's the problem with all pointers tbh as the config is a pointer as well and it could lead to NPE as well. One thing that we can do is, adding validation for all configs

Reply:
How? the `MaxPrice.Value` is string and the `MaxPrice` is of a type `ConfigVarString` which is a struct not a pointer. `SpotInstanceConfig` is a pointer however there is already a nil check before it

Reply:
Oh, my bad. For some reason, I thought those three values are pointers. Ignore my comment.